swabbie-aws/src/test/java/com/netflix/spinnaker/swabbie/aws/images/AmazonImageHandlerTest.kt

```diff
@@ -0,0 +1,501 @@
+
+object AmazonImageHandlerTest {
+  private val front50ApplicationCache = mock<InMemoryCache<Application>>()
+  private val accountProvider = mock<AccountProvider>()
+  private val resourceRepository = mock<ResourceTrackingRepository>()
+  private val resourceOwnerResolver = mock<ResourceOwnerResolver<AmazonImage>>()
+  private val clock = Clock.systemDefaultZone()
+  private val applicationEventPublisher = mock<ApplicationEventPublisher>()
+  private val lockingService = Optional.empty<LockingService>()
+  private val orcaService = mock<OrcaService>()
+  private val imageProvider = mock<ResourceProvider<AmazonImage>>()
+  private val instanceProvider = mock<ResourceProvider<AmazonInstance>>()
+  private val launchConfigurationProvider = mock<ResourceProvider<AmazonLaunchConfiguration>>()
+  private val subject = AmazonImageHandler(
+    clock = clock,
+    registry = NoopRegistry(),
+    notifier = mock(),
+    rules = listOf(OrphanedImageRule()),
+    resourceTrackingRepository = resourceRepository,
+    exclusionPolicies = listOf(
+      LiteralExclusionPolicy(),
+      WhiteListExclusionPolicy(front50ApplicationCache, accountProvider)
+    ),
+    resourceOwnerResolver = resourceOwnerResolver,
+    applicationEventPublisher = applicationEventPublisher,
+
+    lockingService = lockingService,
+    retrySupport = RetrySupport(),
+    imageProvider = imageProvider,
+    instanceProvider = instanceProvider,
+    launchConfigurationProvider = launchConfigurationProvider,
+    orcaService = orcaService,
+    accountProvider = accountProvider
+  )
+
+  @BeforeEach
+  fun setup() {
+    whenever(front50ApplicationCache.get()) doReturn
+      setOf(
+        Application(name = "testapp", email = "name@netflix.com"),
+        Application(name = "important", email = "test@netflix.com"),
+        Application(name = "random", email = "random@netflix.com")
+      )
+
+    whenever(accountProvider.getAccounts()) doReturn
+      setOf(
+        SpinnakerAccount(
+          name = "test",
+          accountId = "1234",
+          type = "aws",
+          edda = "http://edda",
+          regions = listOf(Region(name = "us-east-1")),
+          eddaEnabled = false
+        ),
+        SpinnakerAccount(
+          name = "prod",
+          accountId = "4321",
+          type = "aws",
+          edda = "http://edda",
+          regions = listOf(Region(name = "us-east-1")),
+          eddaEnabled = false
+        )
+      )
+  }
+
+  @AfterEach
+  fun cleanup() {
+    validateMockitoUsage()
+    reset(imageProvider, accountProvider, instanceProvider, launchConfigurationProvider, applicationEventPublisher)
+  }
+
+  @Test
+  fun `should handle work for images`() {
+    Assertions.assertTrue(subject.handles(getWorkConfiguration()))
+  }
+
+  @Test
+  fun `should find image cleanup candidates`() {
+    val params = Parameters(mapOf("account" to "1234", "region" to "us-east-1"))
+    whenever(imageProvider.getAll(params)) doReturn listOf(
+      AmazonImage(
```

Comment:
Would be helpful if you could pull this `whenever(imageProvider.getAll(params)) do return listOf(...)` to the top so that it could be used in every test w/o having to repeat it so much? Nitting just for ease of understanding the tests/making more.